dhis-2/dhis-api/src/main/java/org/hisp/dhis/common/QueryFilter.java

```diff
@@ -39,16 +44,16 @@
 {
     public static final String OPTION_SEP = ";";
 
-    public static final ImmutableMap<QueryOperator, String> OPERATOR_MAP = ImmutableMap
-        .<QueryOperator, String> builder()
-        .put( QueryOperator.EQ, "=" )
-        .put( QueryOperator.GT, ">" )
-        .put( QueryOperator.GE, ">=" )
-        .put( QueryOperator.LT, "<" )
-        .put( QueryOperator.LE, "<=" )
-        .put( QueryOperator.NE, "!=" )
-        .put( QueryOperator.LIKE, "like" )
-        .put( QueryOperator.IN, "in" ).build();
+    public static final ImmutableMap<QueryOperator, Function<Boolean, String>> OPERATOR_MAP = ImmutableMap
+        .<QueryOperator, Function<Boolean, String>> builder()
+        .put( QueryOperator.EQ, isValueNull -> isValueNull ? "is" : "=" )
+        .put( QueryOperator.GT, unused -> ">" )
```

Comment:
Do you think you could explain what is going on here?

Reply:
QueryFilter is used to transform an Operator into part of an SQL condition.
So for instance EQ gets transformed into "=", GT into > and so on.
Now we introduced the ability to query for null values as well, so that translates in SQL into "field is null".
So the right side of the map is a function that translates, for each operator, a boolean into a SQL operator. The boolean represents whether the value on which to apply  the operator is null or not, in fact, what we want to achieve here is that:
value:X --> field = X
value null --> field is null
Since it makes sense to have conditions involving null only for EQ, NE, and IN operators, with remaining operators no transformation occur.